EIPS/eip-7541.md

```diff
@@ -0,0 +1,139 @@
+
+```python
+def execute_block(block: Block) -> None:
+	...
+	coinbase_reward = 0
+
+	for tx in block.transactions:
+		signer(tx).balance -= tx.fee
+
+		# execute transaction
+		...
+
+		signer(tx).balance += refund
+		coinbase_reward += tx.fee - refund 
+
+	# apply coinbase reward at the end of the block
+	coinbase(block).balance += coinbase_reward
+```
+
+## Rationale
+
+This EIP removes the inherent dependency between transactions within a single block. By aggregating fees and applying them at the end, transactions might be executed in parallel without waiting for the previous ones to complete.
+
+## Backwards Compatibility
+
+Balance is only incremented once at the end of the block and this means that `COINBASE` will not have it incremented within this block.
+
+This EIP affects block building and transaction bundles. The delay of `COINBASE` rewards changes their mechanisms.
+
+### Direct Coinbase Payments
+
+This change does not impact direct payments to `COINBASE`. They can still occur alongside the transaction fees.
+
+### Block Building
+
+**Block Value**
+
+This change will require block builders to accumulate transaction fees when evaluating block value. It does not have a fundamental impact on the profitability of the block.
+
+**Block Payout**
+
+This change imposes an additional constraint on the builder’s balance to be able to pay for the block they build before receiving the reward as a refund. The effects of this change are discussed in Security Consideration section in more detail.
+
+### Transaction Bundles
+
+The changes to the timing of fee application are only noticeable across transactions. Transaction bundles that rely on coinbase reward being applied at the end of each transaction might need to adjust the logic.
+
+## Test Cases
+
+**Test Case #1**
+
+```
+Sender A Balance Before Block: 1 ETH
+Coinbase Balance Before Block: 10 ETH
+
+Transaction #1 from Sender A
+Transaction Fee: 0.1 ETH
+Sender A Balance After Transaction #1: 0.9 ETH
+Coinbase Balance After Transaction #1: 10 ETH
+
+Transaction #2 from Sender A
+Transaction Fee: 0.2 ETH 
+Sender A Balance After Transaction #2: 0.7 ETH
+Coinbase Balance After Transaction #2: 10 ETH
+
+Sender A Balance After Block: 0.7 ETH
+Coinbase Balance After Block: 10.3 ETH
+```
+
+**Test Case #2**
+
+```
+Sender A Balance Before Block: 1 ETH
+Coinbase Balance Before Block: 10 ETH
+
+Transaction #1 from Sender A
+Transaction Fee: 0.1 ETH
+Sender A Balance After Transaction #1: 0.9 ETH
+Coinbase Balance After Transaction #1: 10 ETH
+
+Transaction #2 from Sender A
+Transaction Fee: 0.2 ETH 
+Sender A Balance After Transaction #2: 0.7 ETH
+Coinbase Balance After Transaction #2: 10 ETH
+
+Sender A Balance After Block: 0.7 ETH
+Coinbase Balance After Block: 10.3 ETH
+```
+
+## Security Considerations
+
+### Double Spending
+
+The sender is still credited at the start of the transaction and receives the refund at the end to avoid double spending.
```

Comment:
```suggestion
The sender is still charged at the start of the transaction and receives the refund at the end to avoid double spending.
```
and here